astroquery/esasky/core.py

```diff
@@ -1492,22 +1492,22 @@ def _get_maps_for_mission(self, maps_table, mission, download_dir, cache, json,
                             with ZipFile(file=BytesIO(response.content)) as zip:
                                 for info in zip.infolist():
                                     if self._ends_with_fits_like_extentsion(info.filename):
-                                        maps.append(fits.open(zip.extract(info.filename)))
+                                        maps.append(fits.open(zip.extract(info.filename, path=mission_directory)))
                         else:
                             file_name = self._extract_file_name_from_response_header(response.headers)
-                            if (file_name == ""):
+                            if file_name == "":
                                 file_name = self._extract_file_name_from_url(product_url)
-                            if(file_name.lower().endswith(self.__TAR_STRING)):
+                            if file_name.lower().endswith(self.__TAR_STRING):
                                 with tarfile.open(fileobj=BytesIO(response.content)) as tar:
                                     for member in tar.getmembers():
                                         tar.extract(member, directory_path)
-                                        maps.append(fits.open(directory_path + member.name))
+                                        maps.append(fits.open(os.path.join(directory_path, member.name)))
```

Comment:
Is there a reason not to use `Path` here?
```suggestion
                                        maps.append(fits.open(Path(directory_path, member.name)))
```